airflow/models/deadline.py

```diff
@@ -90,3 +94,86 @@ def _determine_resource() -> tuple[str, str]:
     def add_deadline(cls, deadline: Deadline, session: Session = NEW_SESSION):
         """Add the provided deadline to the table."""
         session.add(deadline)
+
+
+class DeadlineTrigger:
+    """
+    Store the calculation methods for the various SDeadline Alert triggers.
+
+    TODO:  Embetter this docstring muchly.
+
+    usage:
+
+    In the DAG define a deadline as
+
+    deadline=DeadlineAlert(
+        trigger=DeadlineTrigger.DAGRUN_EXECUTION_DATE,
```

Comment:
I think the term `trigger` is overused in Airflow. We should find something else. Maybe `method`? Or something else :)

Reply:
Agree 100%, lots of the naming here is still horrible.  I was calling that an "anchor" at one point and hated that, "trigger" feels better but as you said, it's already overused in Airflow.  I'm hoping a better name will come up once I/we see it in use a bit more than the abstract ideas.  Definitely open to suggestions.